### PR TITLE
Fix wp_load_alloptions() returning the wrong assoc array format.

### PR DESCRIPTION
**Pull request by joehoyle**
wp_load_alloptions() expects the array to be keyed by option names, not cache keys. This breaks deleting options, as WP core expects to be able to unset the deleted option form the alloptions array.

### CODE REVIEW

File: object-cache.php [hunk 1]
```diff
@@ -278,13 +278,20 @@ function wp_cache_fetch_all() {
...
+		$caller = debug_backtrace( DEBUG_BACKTRACE_IGNORE_ARGS );
+		$caller = $caller[0];
+	}
+
+	if ( 'cli' !== php_sapi_name() ) {
 		trigger_error( sprintf( 'wp_cache_flush() is only allowed via WP CLI. Called in %s line %d', $caller['file'], $caller['line'] ), E_USER_WARNING );
 		return false;
 	}
-	trigger_error( sprintf( 'wp_cache_flush() used, this is broadly not recommended. Called in %s line %d', $caller['file'], $caller['line'] ), E_USER_WARNING );
+
```
Comment (rmccue) [inline object-cache.php]: These changes seem unrelated?

File: object-cache.php [hunk 2]
```diff
@@ -1524,6 +1531,10 @@ public function getAllOptions() {
 			return array();
 		}
 
+		// getMulti returns a map of `[ cache_key => value ]` but we need to
+		// return a map of `[ option_name => value ]`
+		$data = array_combine( array_keys( $keys ), array_values( $data ) );
```
Comment (rmccue) [inline object-cache.php]: We should put `array_keys()` on the `$this->get` call to avoid doing it twice. Also, don't _think_ `array_values` is needed.

**Review by rmccue — CHANGES_REQUESTED** [review]

File: object-cache.php (hunk 2 above)
Comment (joehoyle) [reply object-cache.php]: Fixed in b1e7f48

File: object-cache.php (hunk 1 above)
Comment (joehoyle) [reply object-cache.php]: Reverted - still fixed a notice thought.

**Review by rmccue — APPROVED** [review]
